monitoring/defender/autotask/on_module_factory_events.js

```diff
@@ -0,0 +1,1222 @@
+  const ensRegistryContract = new ethers.Contract(
+    ENS_REGISTRY,
+    ABI_REGISTRY,
+    provider
+  );
+  const nameHash = ethers.utils.namehash(ensName);
+  const owner = await ensRegistryContract.owner(nameHash);
+  return ethers.utils.getAddress(address) === ethers.utils.getAddress(owner);
+};
+
+/**
+ * Grab the owner using https://docs.ens.domains/dapp-developer-guide/ens-as-nft#deriving-tokenid-from-ens-name
+ * @param {*} provider Ethers.providers.Provider,
+ * @param {string} ensName Ens name
+ * @param {string} address Avatar address
+ * @param {number} chainId Chain Id
+ * @returns {boolean}
+ */
+const checkIfIsOwner = async (provider, ensName, address, chainId) => {
+  const contract = new ethers.Contract(
+    getBaseRegistrarContractAddress(chainId),
+    ABI_BASE_REGISTRAR,
+    provider
+  );
+  const BigNumber = ethers.BigNumber;
+  const utils = ethers.utils;
+  const name = ensName.replace(".eth", "");
+  const labelHash = utils.keccak256(utils.toUtf8Bytes(name));
+  const tokenId = BigNumber.from(labelHash).toString();
+  const ensOwner = await contract.ownerOf(tokenId);
+  return ethers.utils.getAddress(address) === ethers.utils.getAddress(ensOwner);
+};
+
+/**
+ *
+ * @param {Array} logs //Array of transactions logs
+ * @param {*} utils //EtherJS utils
+ * @returns
+ */
+const decode = async (logs, utils) => {
+  const sig = "ModuleProxyCreation(address,address)"; //ModuleProxyCreation (index_topic_1 address proxy, index_topic_2 address masterCopy)
+  const templateSig = "LogNewTemplate(uint256,address,string)"; //LogNewTemplate (index_topic_1 uint256 template_id, index_topic_2 address user, string question_text)
+  const bytes = utils.toUtf8Bytes(sig);
+  const templateBytes = utils.toUtf8Bytes(templateSig);
+  const keccak = utils.keccak256(bytes);
+  const templateKeccak = utils.keccak256(templateBytes);
+  if (logs) {
+    const newRealityModuleProxies = logs
+      .filter((log) => log.topics !== null)
+      .filter((log) => log.topics[0] === keccak)
+      .map((log) => {
+        const proxy = utils.defaultAbiCoder.decode(
+          ["address"],
+          log.topics[1]
+        )[0];
+        const mastercopy = utils.defaultAbiCoder.decode(
+          ["address"],
+          log.topics[2]
+        )[0];
+        return {
+          proxy: ethers.utils.getAddress(proxy),
+          mastercopy: ethers.utils.getAddress(mastercopy),
+        };
+      })
+      // in tx it can be more than one proxy creation. We need to find only the ones that is the reality module
+      .filter(({ mastercopy }) => mastercopy != REALITY_MODULE_MASTERCOPY)
+      .map(({ proxy }) => proxy);
+
+    const newRealityModuleTemplate = logs
+      .filter((log) => log.topics !== null)
+      .filter((log) => log.topics[0] === templateKeccak);
+
+    const templateQuestionText = utils.defaultAbiCoder.decode(
+      ["string"],
+      newRealityModuleTemplate[0].data
+    );
```

Comment:
We can't reliably use this approach:
1. We don't know if this new question is related to the newly created reality module.
2. Reality Modules can be set up with a preexisting template.

Therefore what we need to do is:
1. Read the associated template id from the newly set up proxy contract (from here: https://github.com/gnosis/zodiac-module-reality/blob/d05a496789079b60125a24f386d1a152e1479f4f/contracts/RealityModule.sol#L36).
2. Look through all LogNewTemplate events emitted by Reality.eth (for mainnet: `0x5b7dD1E86623548AF054A4985F7fc8Ccbb554E2c`). Find the one matching the template id (from 1.), then decode the data.

Reply:
We might be able to use the Etherscan API for this (https://docs.etherscan.io/api-endpoints/logs) 🤔